cdap-docs/reference-manual/source/characters.rst

```diff
@@ -12,10 +12,10 @@ Supported Characters
 
 The Cask Data Application Platform (CDAP) has naming conventions for different components of CDAP.
 
-Streams
--------
+Streams and Metadata
```

Comment:
Seems a little strange that these two are clubbed together. Here is the current character set for various ids in CDAP:
```
private static final CharMatcher namespaceMatcher =
    CharMatcher.inRange('A', 'Z')
    .or(CharMatcher.inRange('a', 'z'))
    .or(CharMatcher.inRange('0', '9'))
    .or(CharMatcher.is('_'));
  // Allow hyphens for other ids.
  private static final CharMatcher idMatcher = namespaceMatcher.or(CharMatcher.is('-'));
  // Allow '.' and '$' for dataset ids since they can be fully qualified class names
  private static final CharMatcher datasetIdCharMatcher = idMatcher.or(CharMatcher.is('.')).or(CharMatcher.is('$'));
```